Source/WebKit/WebProcess/Extensions/WebExtensionContextProxy.cpp

```diff
@@ -43,6 +44,22 @@ void WebExtensionContextProxy::addFrameWithExtensionContent(WebFrame& frame)
     m_extensionContentFrames.add(frame);
 }
 
+std::optional<WebExtensionTabIdentifier> WebExtensionContextProxy::tabIdentifier(WebPage& page) const
+{
+    if (m_popupPageMap.contains(page))
+        return std::get<std::optional<WebExtensionTabIdentifier>>(m_popupPageMap.get(page));
+
+    if (m_tabPageMap.contains(page))
```

Comment:
Is it reasonable to assume that we're more likely to find a match in `m_tabPageMap` for the page and check for that first?

Reply:
I waffled on this. But it is more likely it will be a popup hit, since that is slightly more common than opening an extension URL in a tab.

Reply:
Ah, I didn't realize `m_tabPageMap` contained tabs showing extension content only. I agree with you.